contracts/src/v0.8/dev/automation/test/MercuryRegistry.t.sol

```diff
@@ -0,0 +1,310 @@
+
+  function setUp() public virtual {
+    // Set owner, and fork Arbitrum Goerli Testnet (chain ID 421613).
+    // A public Arbitrum Goerli RPC url is being used, and the fork should be cached in CI so availability is not an issue for test runs.
+    vm.startPrank(OWNER);
+    try vm.envBool("CI") returns (bool /* ci */) {
+      vm.selectFork(vm.createFork("https://goerli-rollup.arbitrum.io/rpc", BLOCK_NUMBER));
+    } catch {
+      vm.selectFork(vm.createFork("https://goerli-rollup.arbitrum.io/rpc"));
+    }
+    vm.chainId(31337); // restore chain Id
+
+    // Use a BTC feed and ETH feed.
+    feedIds = new string[](2);
+    feedIds[0] = s_BTCUSDFeedId;
+    feedIds[1] = s_ETHUSDFeedId;
+
+    // Initialize with BTC feed.
+    string[] memory initialFeedIds = new string[](1);
+    initialFeedIds[0] = feedIds[0];
+    string[] memory initialFeedNames = new string[](1);
+    initialFeedNames[0] = "BTC/USD";
+    s_testRegistry = new MercuryRegistry(
+      initialFeedIds,
+      initialFeedNames,
+      address(0), // verifier unset
+      DEVIATION_THRESHOLD,
+      STALENESS_SECONDS
+    );
+    s_testRegistry.setVerifier(VERIFIER); // set verifier
+
+    // Add ETH feed.
+    string[] memory addedFeedIds = new string[](1);
+    addedFeedIds[0] = feedIds[1];
+    string[] memory addedFeedNames = new string[](1);
+    addedFeedNames[0] = "ETH/USD";
+    s_testRegistry.addFeeds(addedFeedIds, addedFeedNames);
+  }
+
+  function testMercuryRegistry() public {
+    // Check upkeep, receive Mercury revert.
+    uint256 blockNumber = BLOCK_NUMBER;
+    vm.roll(blockNumber);
+    vm.expectRevert(
+      abi.encodeWithSelector(
+        FeedLookupCompatibleInterface.FeedLookup.selector,
+        "feedIdHex", // feedParamKey
+        feedIds, // feed Ids
+        "blockNumber", // timeParamKey
+        blockNumber, // block number on which request is occuring
+        "" // extra data
+      )
+    );
+    s_testRegistry.checkUpkeep("");
+
+    // Obtain mercury report off-chain (for August 22 BTC/USD price)
+    bytes[] memory values = new bytes[](1);
+    values[0] = s_august22BTCUSDMercuryReport;
+
+    // Pass the obtained mercury report into checkCallback, to assert that an update is warranted.
+    (bool shouldPerformUpkeep, bytes memory performData) = s_testRegistry.checkCallback(values, bytes("LOOKUP_DATA"));
```

Comment:
Is `bytes("LOOKUP_DATA")` the actual argument that is passed into checkCallback or is it something else? You don't really use it in the `checkCallback` except for abi-decoding. Wondering what's the significance of it.

Reply:
Yeah this is honestly a misnomer on my part. The calldata in `checkUpkeep` is unused by this contract but it's confusing for me to put "LOOKUP_DATA" here as that's not what it is. Will fix.